app/src/main/java/org/thoughtcrime/securesms/configs/ConfigToDatabaseSync.kt

```diff
@@ -207,15 +210,32 @@ class ConfigToDatabaseSync @Inject constructor(
         } else {
             groupInfoConfig.deleteBefore?.let { removeBefore ->
                 val messages = mmsSmsDatabase.getAllMessageRecordsBefore(threadId, TimeUnit.SECONDS.toMillis(removeBefore))
-                val (controlMessages, visibleMessages) = messages.partition { it.isControlMessage }
+                val (controlMessages, visibleMessages) = messages.map { it.first }.partition { it.isControlMessage }
 
                 // Mark visible messages as deleted, and control messages actually deleted.
                 conversationRepository.markAsDeletedLocally(visibleMessages.toSet(), context.getString(R.string.deleteMessageDeletedGlobally))
                 conversationRepository.deleteMessages(controlMessages.toSet(), threadId)
+
+                // if the current user is an admin of this group they should also remove the message from the swarm
+                // as a safety measure
+                val groupAdminAuth = configFactory.getGroup(groupInfoConfig.id)?.adminKey?.data?.let {
+                    OwnedSwarmAuth.ofClosedGroup(groupInfoConfig.id, it)
+                } ?: return
+
+                // remove messages from swarm SnodeAPI.deleteMessage
+                GlobalScope.launch(Dispatchers.Default) {
+                    val cleanedHashes: List<String> =
+                        messages.map { it.second }.filter { !it.isNullOrEmpty() }.filterNotNull()
```

Comment:
Might be good to use sequence to do these many filtering:

```suggestion
                    val cleanedHashes: List<String> =
                        messages.asSequence().map { it.second }.filter { !it.isNullOrEmpty() }.filterNotNull().toList()
```